xchange-stream-service-netty/src/main/java/info/bitrich/xchangestream/service/netty/NettyStreamingService.java

```diff
@@ -375,16 +374,25 @@ public void sendMessage(String message) {
                     Subscription newSubscription = new Subscription(e, channelName, args);
                     try {
                       sendMessage(getSubscribeMessage(channelName, args));
-                    } catch (IOException throwable) {
-                      e.onError(throwable);
+                    } catch (
+                        Exception
+                            throwable) { // if getSubscribeMessage throws this, it is because it
+                      // needs to report
+                      e.onError(throwable); // a problem creating the message
                     }
                     return newSubscription;
                   });
             })
         .doOnDispose(
             () -> {
               if (channels.remove(channelId) != null) {
-                sendMessage(getUnsubscribeMessage(channelId));
+                try {
+                  sendMessage(getUnsubscribeMessage(channelId));
```

Comment:
sendMessage itself is asynchronous - I've been thinking for a while that it should return a `Completable` (see the draft PR I have open). Is that likely to be a problem for you here? Are you only worried about exceptions from `getUnsubscribeMessage` and `getSubscribeMessage`?

Reply:
> Are you only worried about exceptions from getUnsubscribeMessage and getSubscribeMessage?

Yes, they are overrides, so code is not trustworthy